ls/src/main.rs

```diff
@@ -0,0 +1,351 @@
+use std::{
+    fs,
+    io::{self, Write},
+    os::unix::fs::MetadataExt,
+    path, process,
+    string::String,
+    time::SystemTime,
+};
+
+use pad::{Alignment, PadStr};
+
+extern crate chrono;
+
+mod cli;
+mod file;
+mod flags;
+
+use file::File;
+use flags::Flags;
+
+fn main() -> io::Result<()> {
```

Comment:
I think it would be best to keep the main function without return type, because when you use `?` operator on main, it mus certainly exit with the wrong message (no "ls" at the start of the message, and just the debug print of the error, usually we want the display output), and the exit code usually are wrong too.

I know this is painful, I think that as well LOL
But when and if the [`Termination`](https://doc.rust-lang.org/nightly/std/process/trait.Termination.html) trait stabilizes, our lives can be a lot easier.